src/compile.js

```diff
@@ -525,6 +525,15 @@ Compiler.prototype.ccall = function (e) {
     }
     else {
         out ("$ret;"); // This forces a failure if $ret isn't defined
+        if (Sk.python3 && e.func.id && e.func.id.v === "super" && args.length == 0) {
+            // make sure there is a self variable
+            // note that it's part of the js API spec: https://developer.mozilla.org/en/docs/Web/API/Window/self
+            // so we should probably add self to the mangling
+            // TODO: feel free to ignore the above
+            out("if (typeof self === \"undefined\" || self.toString().indexOf(\"Window\") > 0) { throw new Sk.builtin.RuntimeError(\"super(): no arguments\") };")
+            args.push("self.ob$type");
```

Comment:
I think CPython uses the compile-time context to see the name of the class this is being called from within, not using the run-time type of self.  I can't immediately think of a reason why this would result in different behavior, though.  And it doesn't seem like we have enough context in the compiler at this location.  But, is this definitely equivalent?